app/tests/unit/components/Project/ProjectMilestoneDue.test.tsx

```diff
@@ -0,0 +1,147 @@
+import { screen } from "@testing-library/react";
+import ProjectMilestoneDue from "components/Project/ProjectMilestoneDue";
+import { graphql } from "react-relay";
+import ComponentTestingHelper from "tests/helpers/componentTestingHelper";
+import compliledProjectMilestoneDueQuery, {
+  ProjectMilestoneDueQuery,
+} from "__generated__/ProjectMilestoneDueQuery.graphql";
+
+const testQuery = graphql`
+  query ProjectMilestoneDueQuery($project: ID!) @relay_test_operation {
+    query {
+      project(id: $project) {
+        ...ProjectMilestoneDue_project
+      }
+    }
+  }
+`;
+
+const mockQueryPayload = {
+  Project() {
+    return {
+      latestCommittedProjectRevision: {
+        upcomingReportingRequirementFormChange: {
+          asReportingRequirement: {
+            reportDueDate: "2020-01-01",
+          },
+        },
+      },
+      latestCompletedReportingRequirements: {
+        edges: [
+          {
+            node: {
+              reportDueDate: "2020-01-02",
+            },
+          },
+        ],
+      },
+    };
+  },
+};
+
+const componentTestingHelper =
+  new ComponentTestingHelper<ProjectMilestoneDueQuery>({
+    component: ProjectMilestoneDue,
+    compiledQuery: compliledProjectMilestoneDueQuery,
+    testQuery: testQuery,
+    defaultQueryResolver: mockQueryPayload,
+    getPropsFromTestQuery: (data) => ({ project: data.query.project }),
+  });
+
+describe("The Project Milestone Due component", () => {
+  beforeEach(() => {
+    componentTestingHelper.reinit();
+  });
+
+  it("Displays 'none' if there are no requirements completed nor upcoming", () => {
+    componentTestingHelper.loadQuery({
+      Project() {
+        return {
+          latestCommittedProjectRevision: {
+            upcomingReportingRequirementFormChange: null,
+          },
+          latestCompletedReportingRequirements: {
+            edges: [],
+          },
+        };
+      },
+    });
+    componentTestingHelper.renderComponent();
+
+    expect(screen.getByText("None")).toBeInTheDocument();
+  });
+
+  it("Displays 'Complete' if there are no upcoming reporting requirements, and completed reporting requirements", () => {
+    componentTestingHelper.loadQuery({
+      Project() {
+        return {
+          latestCommittedProjectRevision: {
+            upcomingReportingRequirementFormChange: null,
+          },
+          latestCompletedReportingRequirements: {
+            edges: [
+              {
+                node: {
+                  reportDueDate: "2020-01-02",
```

Comment:
Should testing dates include timestamps?

Reply:
eh, we're not testing the timestamp parsing function here, so for simplicity's sake I left it out